[6.1] Fix TinyMCE editor not respecting width and height settings

RickR2H opened the pull request:
Pull Request for Issue #46320

### Summary of Changes
This update improves the Joomla CMS TinyMCE editor plugin configuration and behavior consistency so the editor will respect the settings in the plugin configuration. The following changes have been implemented:

- Removed editor width on Joomla installation so the editor width now defaults to 100% automatically.
- Set default height to 550px and width to 100% in both the plugin configuration and database during installation for consistent initialization.
- Ensured the editor respects configuration values, making sure width and height are applied according to the settings defined in the plugin parameters.
- Adjusted the editor’s z-index to ensure it properly overlaps the settings panel on the right side when editing articles, preventing visual clipping or overlap issues.
- Added a missing description to the editor plugin configuration for better clarity and documentation within the administrator interface.

### Testing Instructions
1. Install Joomla and verify that in the TinyMCE plugin configuration:
  - The width field is 100%.
  - The height is set to 550px.
  - Build the SCSS via: **npm run build:css**

2. Change the editor width to 90vw in the TinyMCE plugin settings.
  - Open an article for editing and confirm that the editor overlaps the settings panel on the right side.
  - Use the resize grip at the bottom of the editor to verify that it correctly overlays the side panel.

3. Change the editor height to 1000px in the TinyMCE plugin settings.
  - Open an article and confirm that the editor now renders with a height of 1000px.

4. Check the descriptions in the TinyMCE plugin settings and verify that all fields display proper descriptions.

5. Open for example modules\mod_custom\mod_custom.xml and add the following code in the xml after the background image field:

```
<field
	name="editor1"
	type="editor"
	label="text editor 1"
	height="900"
	width="900"
/>
<field
	name="editor2"
	type="editor"
	label="text editor 2"
	height="900"
/>
```
Check in a custom HTML module is the editor fields now below the background image respect the width and height set in the field parameters.

6. Add a custom field with type Editor. Check if the default editor size values are respected (width 100% and height 250px). Change the editor render width and height in the custom fields check if these are also respected.

### Actual result BEFORE applying this Pull Request
- The editor ignores width and height configuration values and always reverts to default settings.
- The editor appears underneath the right-side settings panel when adjusting its width.

### Expected result AFTER applying this Pull Request
- The editor respects width and height values set in the plugin configuration.
- The editor correctly overlaps the right-side settings panel when its width is adjusted.
- All configuration field descriptions are visible and accurate.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

RickR2H committed:
Initial commit to fix editor width and height

RickR2H committed:
Fix lint issue

RickR2H committed:
Added space after class

richard67 commented on installation/sql/mysql/base.sql:
```diff
@@ -285,7 +285,7 @@ INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`,
 (0, 'plg_editors_codemirror', 'plugin', 'codemirror', 'editors', 0, 1, 1, 0, 1, '', '{"lineNumbers":"1","lineWrapping":"1","matchTags":"1","matchBrackets":"1","marker-gutter":"1","autoCloseTags":"1","autoCloseBrackets":"1","autoFocus":"1","theme":"default","tabmode":"indent"}', '', 1, 0),
 (0, 'plg_editors_none', 'plugin', 'none', 'editors', 0, 1, 1, 1, 1, '', '', '', 2, 0),
-(0, 'plg_editors_tinymce', 'plugin', 'tinymce', 'editors', 0, 1, 1, 0, 1, '', '{"configuration":{"toolbars":{"2":{"toolbar1":["bold","underline","strikethrough","|","undo","redo","|","bullist","numlist","|","pastetext"]},"1":{"menu":["edit","insert","view","format","table","tools"],"toolbar1":["bold","italic","underline","strikethrough","|","alignleft","aligncenter","alignright","alignjustify","| ...
+(0, 'plg_editors_tinymce', 'plugin', 'tinymce', 'editors', 0, 1, 1, 0, 1, '', '{"configuration":{"toolbars":{"2":{"toolbar1":["bold","underline","strikethrough","|","undo","redo","|","bullist","numlist","|","pastetext"]},"1":{"menu":["edit","insert","view","format","table","tools"],"toolbar1":["bold","italic","underline","strikethrough","|","alignleft","aligncenter","alignright","alignjustify","| ...
```
Why do you change only the `mysql/base.sql` but not the `postgresql/base.sql`?

RickR2H replied:
Done!

Thanks @richard67

RickR2H committed:
Change default vlaues in postgresql

RickR2H commented:
> ```
> <field type="editor" name="text1" label="Text 1" width="100%" height="200"/>
> <field type="editor" name="text2" label="Text 2" />
> ```

@Fedik I will check! Thanks for the feedback

brianteeman commented:
> @brianteeman is this something we have to do in an other PR? I've never updated TinyMCE, so could use some help if that is necessary,

You would need to check with the maintainers if they will accept an update to TinyMCE in a patch release. I only mentioned it because at least from their changelog it is relevant to this PR. I wouldnt want you to waste time fixing a bug if its already been fixed

RickR2H committed:
Set default width, height in DB
Remove default values from thre editor field
Use TinyMCE params when values are not set.

RickR2H commented:
@Fedik I removed the default values from the EditorField. The default is now set when the editor is generated in thr DisplayTrait.php. Could this be a solution?

RickR2H committed:
Merge branch '6.0-dev' into editor-height

RickR2H committed:
Add the defined width and height settings to the default array

richard67 committed:
Merge branch '6.1-dev' into editor-height

RickR2H committed:
Revert Changes

RickR2H committed:
Merge branch '6.1-dev' into editor-height

RickR2H committed:
Removed getting parameters from plugin settings

Fedik commented:
Now looks good, thanks!

richard67 committed:
Merge branch '6.1-dev' into editor-height

tecpromotion committed:
Merge branch '6.1-dev' into editor-height

RickR2H commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46438">issues.joomla.org/tracker/joomla-cms/46438</a>.</sub>

RickR2H committed:
Merge branch '6.1-dev' into editor-height

tecpromotion committed:
Merge branch '6.1-dev' into editor-height

RickR2H commented:
@brianteeman I've created a fix: #47305
